projects/telemetry/inc/gps.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+// This will be the GPS driver
+#include <stdio.h>
+#include "status.h"
+#include "uart.h"
+#include "nmea.h"
+
+// This enum contains the list of supported NMEA sentences that are supported by our GPS chip
+
+typedef void (*GPSHandler)(const NMEAResult);
+typedef void (*GGAHandler)(const GGASentence);
+
+StatusCode evm_gps_init(UARTSettings* settings);
+
+// These methods will add the handler to the handler array, and returns the index so that it can
+// be removed. If the array is full, then it returns -1.
+
+// The reason for different handlers is so that it is easier to use the driver.
+int32_t add_gps_handler(GPSHandler handler);
```

Comment:
Is there a reason why you aren't using the `status.h` library for these return codes?

Reply:
True, I'll change it when I get the chance

Reply:
It returns the index of the location where the handler is stored, and I don't think that can be done with the `status.h` library.

That being said, I think you can still use that library, but you'll also need to either get rid of the `remove_*_handler` functions, or have them take the handler as a argument instead, and then walk the array to find the matching element to remove.

Reply:
The way I was thinking of doing it is to return a StatusCode, but have the `add_*_handler` functions also require a int pointer as a parameter, and then we can just pass the array index back that way.

Reply:
Done, it's returning status codes now